### PR TITLE
Airlines Base functionality

### PR DESCRIPTION
**Pull request by Light-it-luc**
# Light-it Pull Request ⚡⚡⚡

## Description ⭐

Implement the CRUD functionality of the Airlines (new Airline controller class, airlines view and changes in routes) using the JavaScript API instead of jQuery. Updated the blade components for improved flexibility and updated the cities view to implement the new templates.

Simplified modal logic in cities view (instead of having two modals - one for errors and one for editing - modified code to have only one code, and display error messages below inputs when editing). Also upated many HTML selectors (removed most custom HTML attributes and used ids instead, as the  custom attributes were being used only once in all cases).

### Type of change 🏃

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### Requires ❗

- [x] This change requires a documentation update

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g., Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!
> 🟡 Why was the default value removed?

E.g., Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

|              |                |                                     |
| ------------ | -------------- | ----------------------------------- |
| Blocking     | 🔴 ❌ 🚨       | RED                                 |
| Non-blocking | 🟡 💡 🤔 💭    | Yellow, thinking, etc               |
| Praise       | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

## Links

- [Git Flow](https://lightit.slite.com/app/docs/SC8usN2Ju)
- [Handbook of good practices for reviewers in Code Reviews](https://lightit.slite.com/app/docs/ddNGohWthVB3fO)

### CODE REVIEW

File: resources/views/airlines.blade.php [hunk 1]
```diff
@@ -0,0 +1,311 @@
+@php
+    $title = 'Airlines';
+    $columnTitles = ['ID', 'Name', 'Description', 'Flights'];
+@endphp
```
Comment (ezef) [inline resources/views/airlines.blade.php]: Usually we do not want to put PHP code into Blade, cant you pass this directly on the attributes?

File: resources/views/airlines.blade.php [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+                const content = parseErrors(errors)
+                return displayModal("Creation Failed", content, "", "bg-red-500")
+            }
+
+            name.value = ""
+            description.value = ""
+
+            const storedAirline = await response.json()
+            return insertAirlineRow(storedAirline)
+        }
```
Comment (ezef) [inline resources/views/airlines.blade.php]: would be nice to separate responsabilitites here. 
- get values from input
- make API call
- reset form

File: resources/views/components/navlink.blade.php [hunk 3]
```diff
@@ -1,5 +1,7 @@
-<div class="w-24 ml-2 py-2 px-4 text-center rounded-xl font-semibold
-            text-lg text-gray-700 hover:text-black hover:bg-gray-200">
+<div class="w-24 ml-2 py-2 px-4 text-center rounded-xl font-semibold text-lg
+           {{ (Request::is($attributes['href'])) ?
```
Comment (ezef) [inline resources/views/components/navlink.blade.php]: why are you doing this here?

File: resources/views/airlines.blade.php [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+            </x-table>
+        </div>
+    </div>
+
+    <div class="mt-12 mb-4 px-12">
+        {{ $airlines->links() }}
+    </div>
+
+    <script>
+        function displayModal(title, content, footer="", color="bg-indigo-500") {
```
Comment (guillemartinicore) [inline resources/views/airlines.blade.php]: remember not to use function. instead use 'const ... = () => {}'

Comment (guillemartinicore) [reply resources/views/airlines.blade.php]: apply this all over the script

File: routes/web.php [hunk 5]
```diff
@@ -24,3 +25,13 @@
...
 });
+
+Route::group([
+    'prefix' => 'airlines',
+    'controller' => AirlineController::class
+], function () {
+    Route::get('/', 'index');
+    Route::post('/', 'store');
+    Route::delete('/{airline}', 'destroy');
+    Route::patch('/{airline}', 'update');
```
Comment (guillemartinicore) [inline routes/web.php]: You should be careful here. Using a patch is viable, however, you need to take a close look and decide whether it is convenient for you to change your validations or not.

**Review by guillemartinicore — CHANGES_REQUESTED** [review]

**Review by guillemartinicore — APPROVED** [review]

**Pull request merged**